OMC/Gemini worker task delivery breaks on gitignored .omc inbox paths (respectGitIgnore=true)

## Summary
OMC worker execution with Gemini currently depends on reading a task file from `.omc/state/.../inbox.md`, but `.omc/` is typically gitignored. This creates a fragile integration with Gemini context settings and can fail or degrade when `respectGitIgnore=true`.

In addition, Gemini emits repeated noisy errors from `ImportProcessor` when it encounters email-like tokens such as `@sosmedecins-tpm.fr)` in loaded context/history.

## Why this matters
- OMC worker orchestration may break or become non-deterministic depending on local Gemini config.
- Failures are hard to diagnose because noisy `ImportProcessor` errors mask the actual root cause.
- In multi-agent workflows, this can block task completion and cause unnecessary retries/token burn.

## Environment
- OMC team worker mode (`ccg-svelte5-crossval/worker-2`)
- Command pattern used by OMC:
  - `gemini --yolo -p "Read and execute your task from: .omc/state/team/<team>/workers/<worker>/inbox.md"`
- Gemini CLI: `0.31.0`
- Node: `v25.6.1`
- OS: macOS `26.3 (25D125)`
- Repo setup: `.gitignore` contains `.omc/`

## Observed behavior
1. Gemini repeatedly logs:
```text
[ERROR] [ImportProcessor] Failed to import sosmedecins-tpm.fr): ENOENT: no such file or directory, access '/.../docs/sosmedecins-tpm.fr)'
```
2. Worker can fail or misbehave depending on local Gemini context policy. With `respectGitIgnore=true`, ignored `.omc/` task paths are not reliable.
3. During the same run, Gemini may also fail with unrelated quota 429; this is separate from the integration defect but makes triage harder.

## Reproduction
1. Ensure `.omc/` is ignored by git in project.
2. Set Gemini context filtering to honor gitignore (`respectGitIgnore=true`).
3. Launch OMC Gemini worker with inbox-path prompt (as above).
4. Observe inability/unreliability reading `.omc/.../inbox.md` and/or repeated `ImportProcessor` ENOENT noise.

## Expected behavior
- OMC workers should execute deterministically regardless of gitignored state directories.
- Worker task delivery should not rely on model-side file discovery of ignored paths.
- No spurious `ImportProcessor` attempts on email-like tokens.

## Root cause hypothesis
### OMC side (primary)
OMC passes a file path in prompt and assumes Gemini will resolve/read it from workspace, but the path is under `.omc/` (gitignored) and thus context-sensitive.

### Gemini side (secondary/noise)
Gemini import parsing appears to treat `@...` tokens (including email/domain strings) as import directives, causing ENOENT noise.

## Proposed fix (OMC)
1. **Inline inbox content** (recommended):
   - Read `.omc/.../inbox.md` server-side and pass content via stdin or direct prompt payload.
   - Do not rely on runtime file discovery by Gemini.
2. Add a **preflight readability check** before launching worker:
   - If inbox path cannot be read in current policy context, fail fast with explicit actionable error.
3. Add an explicit mode/flag for transport (e.g. `inbox_transport=inline|path`) with `inline` as default for Gemini.
4. Escape/sanitize `@` tokens in system wrappers where relevant to reduce accidental import parsing.

## Minimal workaround (for users)
- Pipe task content directly:
```bash
cat .omc/state/team/<team>/workers/<worker>/inbox.md \
| gemini --yolo -p "Read the task from stdin and execute it."
```
- Or keep `respectGitIgnore=false` (less safe, less predictable).

## Acceptance criteria
- [ ] OMC Gemini worker works with `respectGitIgnore=true`.
- [ ] OMC worker no longer depends on `.omc` path discovery by Gemini.
- [ ] Clear preflight error if task payload cannot be delivered.
- [ ] Regression test for task delivery under gitignored state path.

## Notes
I can provide sanitized full logs and exact command transcript if needed.

--- a/src/team/__tests__/runtime-prompt-mode.test.ts
+++ b/src/team/__tests__/runtime-prompt-mode.test.ts
@@ -108,7 +108,7 @@ describe('spawnWorkerForTask – prompt mode (Gemini & Codex)', () => {
     setupTaskDir(cwd);
   });
 
-  it('gemini worker launch args include -p flag with inbox path', async () => {
+  it('gemini worker launch args include -p flag with inline task content', async () => {
     const runtime = makeRuntime(cwd, 'gemini');
 
     await spawnWorkerForTask(runtime, 'worker-1', 0);
@@ -122,8 +122,12 @@ describe('spawnWorkerForTask – prompt mode (Gemini & Codex)', () => {
 
     // Should contain -p flag for prompt mode
     expect(launchCmd).toContain("'-p'");
-    // Should contain the inbox path reference
-    expect(launchCmd).toContain('.omc/state/team/test-team/workers/worker-1/inbox.md');
+    // Should contain the task content inline (not a file reference) — #1148
+    expect(launchCmd).toContain('Initial Task Assignment');
+    expect(launchCmd).toContain('Test task');
+    expect(launchCmd).toContain('Do something');
+    // Should NOT reference inbox.md file path in the prompt arg
+    expect(launchCmd).not.toContain('Read and execute your task from:');
 
     rmSync(cwd, { recursive: true, force: true });
   });
@@ -159,7 +163,7 @@ describe('spawnWorkerForTask – prompt mode (Gemini & Codex)', () => {
     rmSync(cwd, { recursive: true, force: true });
   });
 
-  it('codex worker launch args include positional prompt (no -p flag)', async () => {
+  it('codex worker launch args include positional prompt with inline task content (no -p flag)', async () => {
     const runtime = makeRuntime(cwd, 'codex');
 
     await spawnWorkerForTask(runtime, 'worker-1', 0);
@@ -173,8 +177,11 @@ describe('spawnWorkerForTask – prompt mode (Gemini & Codex)', () => {
 
     // Should NOT contain -p flag (codex uses positional argument, not a flag)
     expect(launchCmd).not.toContain("'-p'");
-    // Should contain the inbox path as a positional argument
-    expect(launchCmd).toContain('.omc/state/team/test-team/workers/worker-1/inbox.md');
+    // Should contain the task content inline (not a file reference) — #1148
+    expect(launchCmd).toContain('Initial Task Assignment');
+    expect(launchCmd).toContain('Test task');
+    // Should NOT reference inbox.md file path in the prompt arg
+    expect(launchCmd).not.toContain('Read and execute your task from:');
 
     rmSync(cwd, { recursive: true, force: true });
   });
@@ -192,6 +199,90 @@ describe('spawnWorkerForTask – prompt mode (Gemini & Codex)', () => {
     );
     // Only one send-keys call: the launch command itself
     expect(sendKeysCalls.length).toBe(1);
+
+    rmSync(cwd, { recursive: true, force: true });
+  });
+});
+
+describe('spawnWorkerForTask – gitignore bypass (#1148)', () => {
+  let cwd: string;
+
+  beforeEach(() => {
+    tmuxCalls.args = [];
+    cwd = mkdtempSync(join(tmpdir(), 'runtime-gitignore-bypass-'));
+    setupTaskDir(cwd);
+  });
+
+  it('prompt-mode agents receive full task content inline, not a file reference', async () => {
+    // When .omc/ is gitignored and Gemini has respectGitIgnore=true,
+    // the worker cannot read .omc/state/.../inbox.md. The fix inlines
+    // the task content directly into the CLI prompt argument.
+    const runtime = makeRuntime(cwd, 'gemini');
+
+    await spawnWorkerForTask(runtime, 'worker-1', 0);
+
+    const launchCall = tmuxCalls.args.find(
+      args => args[0] === 'send-keys' && args.includes('-l')
+    );
+    expect(launchCall).toBeDefined();
+    const launchCmd = launchCall![launchCall!.length - 1];
+
+    // The prompt must contain the actual task description, not a file path
+    expect(launchCmd).toContain('Do something');
+    expect(launchCmd).toContain('done.json');
+    // Must not tell the worker to read from the gitignored inbox path
+    expect(launchCmd).not.toContain('Read and execute your task from:');
+    expect(launchCmd).not.toMatch(/inbox\.md/);
+
+    rmSync(cwd, { recursive: true, force: true });
+  });
+
+  it('inbox.md is still written for debugging even when content is inlined', async () => {
+    const runtime = makeRuntime(cwd, 'gemini');
+
+    await spawnWorkerForTask(runtime, 'worker-1', 0);
+
+    // Inbox file should still exist for debugging/reference
+    const inboxPath = join(cwd, '.omc/state/team/test-team/workers/worker-1/inbox.md');
+    const content = readFileSync(inboxPath, 'utf-8');
+    expect(content).toContain('Initial Task Assignment');
+    expect(content).toContain('Test task');
+
+    rmSync(cwd, { recursive: true, force: true });
+  });
+
+  it('interactive agents (claude) still reference inbox via tmux send-keys', async () => {
+    // Claude does not have respectGitIgnore, so the interactive path
+    // continues to use the file reference via tmux send-keys.
+    const runtime = makeRuntime(cwd, 'claude');
+
+    await spawnWorkerForTask(runtime, 'worker-1', 0);
+
+    // Interactive mode sends "Read and execute your task from: ..." via send-keys
+    const sendKeysCalls = tmuxCalls.args.filter(
+      args => args[0] === 'send-keys' && args.includes('-l')
+    );
+    // First call is launch command, second is the inbox path notification
+    const inboxNotification = sendKeysCalls.find(args =>
+      args.some(a => typeof a === 'string' && a.includes('Read and execute your task from:'))
+    );
+    expect(inboxNotification).toBeDefined();
+
+    rmSync(cwd, { recursive: true, force: true });
+  });
+
+  it('inline prompt contains done.json path so worker can signal completion', async () => {
+    const runtime = makeRuntime(cwd, 'gemini');
+
+    await spawnWorkerForTask(runtime, 'worker-1', 0);
+
+    const launchCall = tmuxCalls.args.find(
+      args => args[0] === 'send-keys' && args.includes('-l')
+    );
+    const launchCmd = launchCall![launchCall!.length - 1];
+
+    // Done signal path must be present so the worker can write completion
+    expect(launchCmd).toContain('.omc/state/team/test-team/workers/worker-1/done.json');
 
     rmSync(cwd, { recursive: true, force: true });
   });

--- a/src/team/runtime.ts
+++ b/src/team/runtime.ts
@@ -597,10 +597,13 @@ export async function spawnWorkerForTask(
     cwd: runtime.cwd,
   });
 
-  // For prompt-mode agents (e.g. Gemini Ink TUI), pass instruction via CLI
-  // flag so tmux send-keys never needs to interact with the TUI input widget.
+  // For prompt-mode agents (e.g. Gemini Ink TUI), pass the full task
+  // instruction inline via CLI flag. This bypasses gitignore restrictions
+  // (e.g. Gemini's respectGitIgnore=true ignoring .omc/ paths) by embedding
+  // the task content directly in the CLI argument instead of referencing an
+  // inbox file that may be invisible to the worker. See #1148.
   if (usePromptMode) {
-    const promptArgs = getPromptModeArgs(agentType, `Read and execute your task from: ${relInboxPath}`);
+    const promptArgs = getPromptModeArgs(agentType, instruction);
     launchArgs.push(...promptArgs);
   }
 

